### PR TITLE
Fixing the versions numbers and links in the site docs

### PR DESCRIPTION
**Pull request by ChethanUK**
Fixing the versions numbers and links in the site docs
- Fixed Bazel and Heron Versions in all versioned docs
- Fixed en language-based problems
- Fixed Release Notes problems
- Fixed Broken Links
- Updated the scripts to build the site

To Test:
```
cd incubator-heron/website2/website
make buildsite
cd ${ROOT_DIR}/generated-site/content/
Use now (npm i -g now & now)
```

### CODE REVIEW

File: website2/website/scripts/build-site.sh [hunk 1]
```diff
@@ -30,5 +30,5 @@ yarn build
 node ./scripts/replace.js
 
 rm -rf ${ROOT_DIR}/generated-site/content
-mkdir -p ${ROOT_DIR}/generated-site/content
+mkdir -p ${ROOT_DIR}/generatedori-site/content
```
Comment (nwangtw) [inline website2/website/scripts/build-site.sh]: this is expected?

File: website2/website/scripts/replace.js [hunk 2]
```diff
@@ -30,14 +29,37 @@ replace(options)
...
 
+
 const versions = getVersions();
 
 const latestVersion = versions[0];
 
+const bazelVersions = {
+    '0.20.0': '0.14.1',
+    '0.20.1': '0.26',
+    '0.20.2': '0.26',
```
Comment (nwangtw) [inline website2/website/scripts/replace.js]: Suggest removing 0.20.2.  Or maybe you mean "latest"?

Comment (ChethanUK) [reply website2/website/scripts/replace.js]: Okay!.

File: website2/website/scripts/build-site.sh (hunk 1 above)
Comment (ChethanUK) [reply website2/website/scripts/build-site.sh]: My Bad, Mistake. I was testing around.. Fixed it..

File: website2/docs/getting-started-local-single-node.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ The easiest way to get started learning Heron is to install the Heron client too
 * [MacOS](#macos-homebrew)
 * [Ubuntu >= 14.04](#using-installation-scripts)
 * [CentOS](#using-installation-scripts)
-
+aa
```
Comment (joshfischer1108) [inline website2/docs/getting-started-local-single-node.md]: Did you mean for this `aa` mark?

File: website2/website/README.md [hunk 4]
```diff
@@ -16,6 +16,11 @@
     specific language governing permissions and limitations
     under the License.
 -->
+ perl -pi -e 's/^  */\t/' Makefile
```
Comment (joshfischer1108) [inline website2/website/README.md]: What is the significance of these few lines?

File: website2/website/versioned_docs/version-0.21.2/getting-started-local-single-node.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
+---
+id: version-0.21.2-getting-started-local-single-node
+title: Local (Single Node)
```
Comment (joshfischer1108) [inline website2/website/versioned_docs/version-0.21.2/getting-started-local-single-node.md]: This page should be removed. We do not have a `0.20.2` version het.

File: website2/website/versioned_docs/version-0.21.2/guides-ui-guide.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
+---
+id: version-0.21.2-guides-ui-guide
```
Comment (joshfischer1108) [inline website2/website/versioned_docs/version-0.21.2/guides-ui-guide.md]: This page should be removed.  We do not have a version `0.20.2`

File: website2/website/versioned_docs/version-0.21.2/user-manuals-heron-ui.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+---
+id: version-0.21.2-user-manuals-heron-ui-runbook
+title: Heron UI Runbook
```
Comment (joshfischer1108) [inline website2/website/versioned_docs/version-0.21.2/user-manuals-heron-ui.md]: This page should be removed.

File: website2/website/versions.json [hunk 8]
```diff
@@ -1,3 +1,5 @@
 [
+  "0.21.1",
+  "0.21.2",
```
Comment (joshfischer1108) [inline website2/website/versions.json]: `0.20.2` should be removed.

File: website2/website/release-notes.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+- cleanup README (#3037)
+- Fix Topology-API window documentation (#3039)
+- Nwang/update thirdparty license (#3042)
+- Fix 'Copyright 2015 The Heron Authors' in scripts (#3036)
+
+### 0.21.1-incubating &mdash; 2019-7-20 <a id="0.20.1"></a>
+
+- Example commit of 0.21.1
+
+### 0.21.2-incubating &mdash; 2019-08-27 <a id="0.21.2"></a>
```
Comment (joshfischer1108) [inline website2/website/release-notes.md]: This line referencing `0.20.2` should be removed.

File: website2/website/README.md (hunk 4 above)
Comment (joshfischer1108) [reply website2/website/README.md]: We should also remove the reference to version `0.20.2`

File: website2/Dockerfile [hunk 10]
```diff
@@ -22,5 +22,6 @@ EXPOSE 3000 35729
 COPY ./docs /app/docs
 COPY ./website /app/website
 RUN yarn install
-
+RUN git config user.name "Heron Publish"
```
Comment (joshfischer1108) [inline website2/Dockerfile]: What is the purpose of adding git configs inside the docker container?

File: website2/website/README.md (hunk 4 above)
Comment (ChethanUK) [reply website2/website/README.md]: okay

File: website2/Dockerfile (hunk 10 above)
Comment (ChethanUK) [reply website2/Dockerfile]: When using npm publish gh-pages it was giving errors, so I used that. Will update

**Comment by joshfischer1108** [comment]
@ChethanUK any updates on this PR?

**Comment by nwangtw** [comment]
It seems like @ChethanUK has been pretty busy recently. @joshfischer1108 maybe let's merge and then clean it up?

**Comment by joshfischer1108** [comment]
@nwangtw Sounds good to me.

**Pull request merged**